### PR TITLE
Faces 4.0: Update to 4.0.0-RC2

### LINKED ISSUE #23077: Update Jakarta Faces 4.0 to RC2
4.0.0-RC2 was released today (Oct 24th, 2022).

### PR DESCRIPTION
**Pull request by volosied**
fixes #23077

### CODE REVIEW

**Comment by volosied** [comment]
#build

**Comment by volosied** [comment]
#build

**Comment by volosied** [comment]
#build

File: dev/com.ibm.ws.jsf.2.3_fat/fat/src/com/ibm/ws/jsf23/fat/tests/JSF23CDIGeneralTests.java [hunk 1]
```diff
@@ -218,6 +218,8 @@ public void testCDIManagedProperty() throws Exception {
         String contextRoot = "CDIManagedProperty";
         try (WebClient webClient = new WebClient()) {
             webClient.setAjaxController(new NicelyResynchronizingAjaxController());
+            // Avoid error: message=[missing ; after for-loop condition]
+            webClient.getOptions().setThrowExceptionOnScriptError(false);
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsf.2.3_fat/fat/src/com/ibm/ws/jsf23/fat/tests/JSF23CDIGeneralTests.java:222]: Would you mind opening an issue and linking here so we can check if we can remove this once we pull in the new MyFaces javascript?

Comment (pnicolucci) [reply dev/com.ibm.ws.jsf.2.3_fat/fat/src/com/ibm/ws/jsf23/fat/tests/JSF23CDIGeneralTests.java:222]: Otherwise, these changes look good. Thanks!

Comment (volosied) [reply dev/com.ibm.ws.jsf.2.3_fat/fat/src/com/ibm/ws/jsf23/fat/tests/JSF23CDIGeneralTests.java:222]: I've created https://github.com/OpenLiberty/open-liberty/issues/23135

**Review by pnicolucci — APPROVED** [review]

**Comment by volosied** [comment]
#libby

**Pull request merged**